### ISSUE TITLE
Control permission with M-Community group

### ISSUE THREAD

**Issue by mfldavidson** [tl-its-umich-edu/student_explorer#121]
Right now, when a user is added or removed, we must separately add/remove the user from the M-Community group that we use to communicate with users. This is extremely error-prone and redundant. I would like to simply be able to add/remove users from the M-Community group https://mcommunity.umich.edu/#group/members:Student%20Explorer%20Users in order to control the highest-level access ("can they log in" vs "can they not log in").

I know that AI does this for advisor access to ART so if we need some guidance on implementing this we can probably talk to them.

Here is a flow chart where I'm trying to detail how this should work. This should help us discuss feasibility.
![Diagram](https://user-images.githubusercontent.com/31591389/60599881-32302680-9d7d-11e9-94e2-332ff144b55c.png)

**Comment by jonespm** [comment]
So shibboleth releases the following attributes https://documentation.its.umich.edu/node/262/#autoincommon

I don't believe anything in here easily answers the question in the first box. eduPersonAffiliation is just student/faculty/staff/employee/alum/affiliate/member. eduPersonEntitlement I don't believe contains mCommunity groups. I think we'd need to talk to them.

Most likely what they're doing is making a secondary query to the UM API after login to validate, which is what we were doing at one time. It would be nice if it was all in the authentication request though.

**Comment by jonespm** [comment]
We discussed changing the flowchart on the side if the user is not in the mcommunity group. If the are not in the group don't do anything as part of this process. Implementation of deactivating the user is difficult and not really necessary since there will be many users that are never deactivated through this process. The better change will be to periodically run some SQL script to deactivate users who haven't logged in. The authorized path will allow users access.

**Comment by mfldavidson** [comment]
Testing using my jmaggiex affiliate account

Situations where access should be DENIED (test success = access properly denied, test fail = access improperly granted):
- Tested NOT IN Mcomm group, DEACTIVATED account -- test success
- Tested NOT IN Mcomm group, ACTIVATED account -- test success
- Tested NOT IN Mcomm group, NO account at all -- test success

Situations where access should be GRANTED (test success = access properly granted, test fail = access improperly denied):
- Tested IN Mcomm group, NO account at all -- test success (account created)
- Tested IN Mcomm group, ACTIVATED account -- test success
- Tested IN Mcomm group, DEACTIVATED account -- test success

I also tested what would happen if I had an active session in SE while in the Mcomm group, access granted and all, and I was removed from the Mcomm group. I was still able to navigate around for a bit--successfully accessed /advisors/ and even a search was successful. However when I clicked "Logout" it then took me to Permission Denied. I don't see any reason that revoking access needs to be a RIGHT THIS RED HOT SECOND action, and this testing shows that it will fairly swiftly kick the user out, so I am satisfied with testing.
